readme.md

```diff
@@ -26,49 +26,287 @@ or
 ```
 ___
 ## Usage
-
+ValidateX aims on providing users with built-in decoders and validators to identify the type of the value and validate the value simultaneously. Along with the built-in decoders and validators, you can create custom decoders/validators.
 ___
 
 ## Key Features
+- Functions as decoders/validators
+- Supports Typescript
+- Static type inference
+- Easy to extend
+- Easy to create custom decoders/validators
 
 ___
 
 ## Kitchen Sink
-
+```typescript
+import v, { Typeof } from 'validatex'
+
+const userSchema = v.array(
+  v.object({
+    name: v.string(),
```

Comment:
Can we have some validators too?
e.g. v.string(v.minlength(1))

Reply:
I'll do that.